src/pcp/js/index.js

```diff
@@ -0,0 +1,490 @@
+     */
+    fluid.defaults("gpii.pcp.mainWindow", {
+        gradeNames: ["fluid.viewComponent"],
+        model: {
+            preferences: {
+                sets: [],
+                activeSet: null,
+                settings: []
+            }
+        },
+        selectors: {
+            theme: "#flc-theme",
+            header: "#flc-header",
+            content: "#flc-content",
+            footer: "#flc-footer",
+            settingsList: "#flc-settingsList",
+            heightChangeListener: "#flc-heightChangeListener"
+        },
+        components: {
+            splash: {
+                type: "gpii.pcp.splash",
+                container: "{that}.container",
+                options: {
+                    listeners: {
+                        "{mainWindow}.events.onPreferencesUpdated": {
+                            funcName: "gpii.pcp.splash.toggleSplashWindow",
+                            args: ["{that}", "{mainWindow}.model.preferences.sets"]
+                        }
+                    }
+                }
+            },
+            header: {
+                type: "gpii.pcp.header",
+                container: "{that}.dom.header"
+            },
+            settingsPanel: {
+                type: "gpii.pcp.settingsPanel",
+                container: "{that}.dom.settingsList",
+                createOnEvent: "{mainWindow}.events.onPreferencesUpdated",
+                options: {
+                    model: {
+                        settings: "{mainWindow}.model.preferences.settings"
+                    },
+                    events: {
+                        onSettingAltered: "{mainWindow}.events.onSettingAltered",
+                        onSettingUpdated: "{mainWindow}.events.onSettingUpdated"
+                    }
+                }
+            },
+            footer: {
+                type: "gpii.pcp.footer",
+                container: "{that}.dom.footer"
+            },
+            heightChangeListener: {
+                type: "gpii.pcp.heightChangeListener",
+                container: "{that}.dom.heightChangeListener",
+                options: {
+                    invokers: {
+                        onHeightChanged: "{mainWindow}.onContentHeightChanged"
+                    }
+                }
+            }
+        },
+        modelListeners: {
+            "preferences": "{that}.events.onPreferencesUpdated"
+        },
+        listeners: {
+            "onCreate.addCommunicationChannel": {
+                funcName: "gpii.pcp.addCommunicationChannel",
+                args: ["{that}"]
+            },
+
+            "onSettingAltered.notifyIPC": {
+                funcName: "gpii.pcp.notifySettingUpdate",
+                args: ["{arguments}.0", "{arguments}.1"]
+            }
+        },
+        invokers: {
+            "updatePreferences": {
+                changePath: "preferences",
+                value: "{arguments}.0",
+                source: "outer"
```

Comment:
Very nice use of source tracking here, but do give it a more descriptive name - remember that these names might propagate around an entire design of components linked by relays. Perhaps something like pcp.mainWindow